artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/TransportConstants.java

```diff
@@ -183,6 +187,10 @@
 
    public static final int DEFAULT_TCP_RECEIVEBUFFER_SIZE = 1024 * 1024;
 
+   public static final int DEFAULT_NETTY_WRITE_BUFFER_LOW_WATER_MARK = 32 * 1024;
```

Comment:
In theory the 2 values are not dependent:  TCP will fragment the data if bigger than its MTU (1460 bytes for Ethernet + TCPv4), hence you could have any size for the TCP buffer (>=MTU AFAIK).
The water marks values affect only the buffering performed by Netty and how many times it will call channelWritabilityChanged, dependently by our writing rate and size.
By consequence it will change how many times channel::isWritable will be true.
I've used the doc and http://normanmaurer.me/presentations/2014-twitter-meetup-netty/slides.html#9.0 and http://normanmaurer.me/presentations/2014-twitter-meetup-netty/slides.html#10.0 to have an idea about it.